tests/test_codegen_advanced.cpp

```
@@ -251,6 +251,88 @@ TEST_F(CodeGenTest, LambdaArgTypeError) {
+        "\n"
+        "c = Counter(0)\n"
+        "inc = ():\n"
+        "    c.value = c.value + 1\n"
+        "    print(c.value)\n"
+        "\n"
+        "inc()\n"
+        "print(c.value)";
+    EXPECT_EQ(runSource(src), "1\n0\n");
+}
+
+TEST_F(CodeGenTest, CapturedVarShadowedByForLoopOK) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)